packages/relayer-lib/src/submitter/eth_eureka.rs

```diff
@@ -0,0 +1,283 @@
+//! The `ChainSubmitter` submits txs to [`EthEureka`] based on events from [`CosmosSdk`].
+
+use std::{env, str::FromStr};
+
+use alloy::{
+    primitives::Address,
+    providers::Provider,
+    sol_types::{SolCall, SolValue},
+    transports::Transport,
+};
+use anyhow::Result;
+use futures::future;
+use ibc_core_host_types::identifiers::ChainId;
+use ibc_eureka_solidity_types::{
+    ics02::client::clientInstance,
+    ics26::{
+        router::{
+            ackPacketCall, multicallCall, recvPacketCall, routerCalls, routerInstance,
+            timeoutPacketCall,
+        },
+        IICS02ClientMsgs::Height,
+        IICS26RouterMsgs::{MsgAckPacket, MsgRecvPacket, MsgTimeoutPacket},
+    },
+};
+// Re-export the `SupportedProofType` enum.
+pub use sp1_ics07_tendermint_prover::prover::SupportedProofType;
+use sp1_ics07_tendermint_prover::{
+    programs::UpdateClientAndMembershipProgram, prover::SP1ICS07TendermintProver,
+};
+
+use sp1_ics07_tendermint_solidity::{
+    sp1_ics07_tendermint,
+    IICS07TendermintMsgs::ClientState,
+    IMembershipMsgs::{MembershipProof, SP1MembershipAndUpdateClientProof},
+    ISP1Msgs::SP1Proof,
+};
+use sp1_ics07_tendermint_utils::{
+    light_block::LightBlockExt, merkle::convert_tm_to_ics_merkle_proof, rpc::TendermintRpcExt,
+};
+use sp1_sdk::HashableKey;
+use tendermint_rpc::{Client, HttpClient};
+
+use crate::{
+    chain::{CosmosSdk, EthEureka},
+    events::EurekaEvent,
+};
+
+use super::r#trait::ChainSubmitterService;
+
+/// The `ChainSubmitter` produces txs to [`EthEureka`] based on events from [`CosmosSdk`].
+#[allow(dead_code)]
+pub struct ChainSubmitter<T: Transport + Clone, P: Provider<T> + Clone> {
+    /// The IBC Eureka router instance.
+    pub ics26_router: routerInstance<T, P>,
+    /// The HTTP client for the Cosmos SDK.
+    pub tm_client: HttpClient,
+    /// The proof type to use for [`SP1ICS07TendermintProver`].
+    pub proof_type: SupportedProofType,
+    /// The SP1 private key for the prover network
+    /// Uses the local prover if not set
+    pub sp1_private_key: Option<String>,
+}
+
+impl<T: Transport + Clone, P: Provider<T> + Clone> ChainSubmitter<T, P> {
+    /// Create a new `ChainSubmitter` instance.
+    pub fn new(
+        ics26_address: Address,
+        provider: P,
+        tm_client: HttpClient,
+        proof_type: SupportedProofType,
+        sp1_private_key: Option<String>,
```

Comment:
Since this is going to be an env var anyway, maybe we could avoid having to deal with sending this around and storing it in config files. Keeping private keys in config files seems like something we might want to avoid.

Reply:
The relayer should not use env variables. Sp1 should switch to api keys perhaps. I view this as an API key anyway. So this is fine